admin/class-admin-init.php

```diff
@@ -42,32 +44,54 @@ public function __construct() {
 	}
 
 	/**
-	 * Redirect first time or just upgraded users to the about screen.
+	 * For WP versions older than 4.2, this includes styles and a script to make notices dismissible.
 	 */
-	public function redirect_to_about_page() {
-		if ( ! current_user_can( 'manage_options' ) ) {
-			return;
+	public function enqueue_dismissible() {
+		if ( version_compare( $GLOBALS['wp_version'], '4.2', '<' ) ) {
+			wp_enqueue_style( 'wpseo-dismissible', plugins_url( 'css/wpseo-dismissible' . WPSEO_CSSJS_SUFFIX . '.css', WPSEO_FILE ), array(), WPSEO_VERSION );
+			wp_enqueue_script( 'wpseo-dismissible', plugins_url( 'js/wp-seo-dismissible' . WPSEO_CSSJS_SUFFIX . '.js', WPSEO_FILE ), array( 'jquery' ), WPSEO_VERSION, true );
 		}
+	}
+	/**
+	 * Redirect first time or just upgraded users to the about screen.
+	 */
+	public function after_update_notice() {
+		if ( current_user_can( 'manage_options' ) && ! $this->seen_about() ) {
 
-		if ( $this->options['seen_about'] ) {
-			return;
-		}
+			if ( filter_input( INPUT_GET, 'intro' ) === '1' ) {
+				update_user_meta( get_current_user_id(), 'wpseo_seen_about_version' , WPSEO_VERSION );
 
-		if ( in_array( $this->pagenow, array(
-			'update.php',
-			'update-core.php',
-			'plugins.php',
-			'plugin-install.php',
-		) ) ) {
-			return;
+				return;
+			}
+
+			$info_message = sprintf(
+				__( 'WordPress SEO by Yoast has been updated to version %1$s. %2$sClick here%3$s to find out what\'s new!', 'wordpress-seo' ),
+				WPSEO_VERSION,
+				'<a href="' . admin_url( 'admin.php?page=wpseo_dashboard&intro=1' ) . '">',
+				'</a>'
+			);
+
+			Yoast_Notification_Center::get()->add_notification( new Yoast_Notification( $info_message, 'updated notice is-dismissible wpseo-dismiss-about', wp_create_nonce( 'wpseo-dismiss-about' ) ) );
 		}
+	}
 
-		// We're redirecting the user to the about screen, let's make sure we don't do it again until he/she upgrades again
-		$wpseo_option               = get_option( 'wpseo' );
-		$wpseo_option['seen_about'] = true;
-		update_option( 'wpseo', $wpseo_option );
+	/**
+	 * Helper to verify if the current user has already seen the about page for the current version
+	 *
+	 * @return bool
+	 */
+	private function seen_about() {
+		return get_user_meta( get_current_user_id(), 'wpseo_seen_about_version', true ) === WPSEO_VERSION;
+	}
 
-		wp_safe_redirect( admin_url( 'admin.php?page=wpseo_dashboard&intro=1' ) );
+	/**
+	 * Helper to verify if the user is currently visiting one of our admin pages.
+	 *
+	 * @return bool
+	 */
+	private function on_wpseo_admin_page() {
+		$page = filter_input( INPUT_GET, 'page' );
```

Comment:
Why assigning this to a var? The value is used only once, I might do this directly in the comparison.